rustls/src/common_state.rs

```diff
@@ -682,6 +695,32 @@ impl CommonState {
     }
 }
 
+/// Describes which sort of handshake happened.
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum HandshakeDetails {
```

Comment:
I don't feel strongly, but I wonder if given the current granularity of information if this would be better named `HandshakeKind`?

Reply:
Yeah I like `HandshakeKind`, will go with that. The best other option I came up with was `HandshakeShape`, but it's a tongue-twister.